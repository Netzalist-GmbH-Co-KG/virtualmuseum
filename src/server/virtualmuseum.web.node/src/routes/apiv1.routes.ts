import express, { Router } from 'express';
import { DatabaseService } from '../services/database.service';
import { MultiMediaPresentationController } from '../controllers/multimediapresentation.controller';
import { TenantController } from '../controllers/tenant.controller';
import { TimeSeriesController } from '../controllers/timeseries.controller';
import { TopographicalTableController } from '../controllers/topographicaltable.controller';


export class ApiV1Router {
    private router: Router;
    private mmpController: MultiMediaPresentationController;
    private tenantController: TenantController;
    private timeSeriesController: TimeSeriesController;
    private topographicalTableController: TopographicalTableController;

    constructor(dbService: DatabaseService) {
        this.router = express.Router();
        this.mmpController = new MultiMediaPresentationController(dbService);
        this.tenantController = new TenantController(dbService);
        this.timeSeriesController = new TimeSeriesController(dbService);
        this.topographicalTableController = new TopographicalTableController(dbService);
        this.setupRoutes();
    }

    private setupRoutes(): void {
        this.router.get('/multimediapresentations/', this.mmpController.getAllMultimediaPresentations.bind(this.mmpController));
        this.router.get('/tenants/', this.tenantController.getAllTenants.bind(this.tenantController));
        this.router.get('/timeseries/', this.timeSeriesController.getTimeSeries.bind(this.timeSeriesController));
        this.router.get('/topographicaltables/:tableId', this.topographicalTableController.getTopographicalTable.bind(this.topographicalTableController));
    }

    getRouter(): Router {
        return this.router;
    }
}
