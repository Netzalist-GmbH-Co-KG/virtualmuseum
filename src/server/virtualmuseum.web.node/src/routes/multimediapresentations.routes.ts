import express, { Router } from 'express';
import { DatabaseService } from '../services/database.service';
import { MultiMediaPresentationController } from '../controllers/multimediapresentation.controller';


export class MultiMediaPresentationRouter {
    private router: Router;
    private controller: MultiMediaPresentationController;

    constructor(dbService: DatabaseService) {
        this.router = express.Router();
        this.controller = new MultiMediaPresentationController(dbService);
        this.setupRoutes();
    }

    private setupRoutes(): void {
        this.router.get('/', this.controller.getAllMultimediaPresentations.bind(this.controller));
    }

    getRouter(): Router {
        return this.router;
    }
}
