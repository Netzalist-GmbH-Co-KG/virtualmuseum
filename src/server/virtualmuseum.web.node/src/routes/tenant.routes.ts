import express, { Router, Request, Response, NextFunction } from 'express';
import { DatabaseService } from '../services/database.service';

export class TenantRouter {
    private router: Router;
    private dbService: DatabaseService;

    constructor(dbService: DatabaseService) {
        this.router = express.Router();
        this.dbService = dbService;
        this.setupRoutes();
    }

    private setupRoutes(): void {
        this.router.get('/', this.getTenants.bind(this));
    }

    private async getTenants(req: Request, res: Response, next: NextFunction): Promise<void> {
        try {
            console.log('Getting tenants...');
            const tenants = await this.dbService.getTenants();
            console.log('Retrieved tenants:', tenants);
            res.json(tenants);
        } catch (error) {
            console.error('Error getting tenants:', error);
            next(error);
        }
    }

    getRouter(): Router {
        return this.router;
    }
}
