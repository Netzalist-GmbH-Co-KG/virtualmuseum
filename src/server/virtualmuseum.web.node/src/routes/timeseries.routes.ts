import express, { Router } from 'express';
import { DatabaseService } from '../services/database.service';
import { TimeSeriesController } from '../controllers/timeseries.controller';


export class TimeSeriesRouter {
    private router: Router;
    private controller: TimeSeriesController;

    constructor(dbService: DatabaseService) {
        this.router = express.Router();
        this.controller = new TimeSeriesController(dbService);
        this.setupRoutes();
    }

    private setupRoutes(): void {
        this.router.get('/', this.controller.getTimeSeries.bind(this.controller));
    }

    getRouter(): Router {
        return this.router;
    }
}
