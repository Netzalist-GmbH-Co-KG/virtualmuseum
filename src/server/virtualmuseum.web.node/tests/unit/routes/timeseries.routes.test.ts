import request from 'supertest';
import express from 'express';
import { TestDatabaseService } from '../../helpers/test-database';
import { ApiV1Router } from '../../../src/routes/apiv1.routes';

describe('TimeSeriesRouter', () => {
    let app: express.Application;
    let dbService: TestDatabaseService;

    beforeAll(async () => {
        dbService = new TestDatabaseService();
        await dbService.initialize();
        console.log('Test setup complete');
    });

    beforeEach(async () => {
        app = express();
        app.use(express.json());
        const apiV1Router = new ApiV1Router(dbService);
        app.use('/api/v1/', apiV1Router.getRouter());
        await dbService.reset();
    });

    afterAll(async () => {
        await dbService.close();
        console.log('Test cleanup complete');
    });

    describe('GET /api/v1/timeseries', () => {
        it('should return all time series with their geo event groups and events', async () => {
            const response = await request(app).get('/api/v1/timeseries');
            
            expect(response.status).toBe(200);
            expect(Array.isArray(response.body)).toBe(true);

            const timeSeries = response.body[0];
            expect(timeSeries).toMatchSnapshot();
        });

        it('should return an empty array when no time series exist', async () => {
            await dbService.clearData();
            const response = await request(app).get('/api/v1/timeseries');
            
            expect(response.status).toBe(200);
            expect(response.body).toEqual([]);
        });
    });
});
