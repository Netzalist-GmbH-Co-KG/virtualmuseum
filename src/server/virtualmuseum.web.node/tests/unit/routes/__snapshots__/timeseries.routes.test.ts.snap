// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSeriesRouter GET /api/v1/timeseries should return all time series with their geo event groups and events 1`] = `
{
  "Description": "Test Time Series Description 1",
  "GeoEventGroups": [
    {
      "Description": "Test Group Description 1",
      "GeoEvents": [
        {
          "DateTime": "2023-01-01T00:00:00Z",
          "Description": "Test Event Description 1",
          "GeoEventGroupId": "test-group-1",
          "Id": "test-event-1",
          "Label": "Test Event 1",
          "Latitude": 0,
          "Longitude": 0,
          "MultimediaPresentationId": "test-presentation-1",
        },
      ],
      "Id": "test-group-1",
      "Label": "Test Group 1",
      "TimeSeriesId": "test-timeseries-1",
    },
  ],
  "Id": "test-timeseries-1",
  "Name": "Test Time Series 1",
}
`;
